mlrun/model_monitoring/controller.py

```diff
@@ -350,25 +350,29 @@ def _delete_old_parquet(
 
         :param endpoints: List of dictionaries of model endpoints records.
         """
+        if self.parquet_directory.startswith("v3io:///"):
 
-        target = ParquetTarget(path=self.parquet_directory)
-        fs = target._get_store().get_filesystem()
+            target = mlrun.datastore.targets.BaseStoreTarget(
+                path=self.parquet_directory
+            )
+            store, _ = target._get_store_and_path()
+            fs = store.get_filesystem()
 
-        # calculate time threshold (keep only files from the last 24 hours)
-        time_to_keep = float(
-            (datetime.datetime.now() - datetime.timedelta(days=days)).strftime("%s")
-        )
-        for endpoint in endpoints:
-            parquet_files = fs.listdir(
-                path=f"{self.parquet_directory}"
-                f"/key={endpoint[mlrun.common.schemas.model_monitoring.EventFieldType.UID]}"
+            # calculate time threshold (keep only files from the last 24 hours)
+            time_to_keep = float(
+                (datetime.datetime.now() - datetime.timedelta(days=days)).strftime("%s")
             )
-            for file in parquet_files:
-                if file["mktime"] < time_to_keep:
-                    # Delete files
-                    fs.rm(path=file["name"], recursive=True)
-                    # Delete directory
-                    fs.rmdir(path=file["name"])
+            for endpoint in endpoints:
+                parquet_files = fs.listdir(
+                    path=f"{self.parquet_directory}"
+                    f"/key={endpoint[mlrun.common.schemas.model_monitoring.EventFieldType.UID]}"
+                )
+                for file in parquet_files:
+                    if file["mtime"] < time_to_keep:
```

Comment:
why not iterate on the dirnames to understand the processing time of each?
also, if you want, you can't assert that file times are what you expect (and log an error if they don't) but we should still delete - otherwise we will also fail deleting the dir and we are "leaking" files.